src/script/components/Modals/FileHistoryModal/FileHistoryModal.styles.ts

```diff
@@ -0,0 +1,226 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CSSObject} from '@emotion/react';
+
+export const fileHistoryModalWrapperCss: CSSObject = {
+  overflow: 'unset',
+  overflowY: 'unset',
+  width: '500px',
+  maxHeight: '650px',
+  height: '80vh',
+  margin: '1rem',
+  borderRadius: '10px',
+};
+
+export const fileVersionRestoreModalWrapperCss: CSSObject = {
+  overflow: 'unset',
+  overflowY: 'unset',
+  width: '460px',
+  height: 'auto',
+  margin: '1rem',
+  borderRadius: '10px',
+};
+
+// Header styles
+export const fileHistoryHeaderContainerCss: CSSObject = {
+  display: 'flex',
+  alignItems: 'end',
+  justifyContent: 'space-between',
+  padding: '12px 12px 12px 16px',
+  height: '100px',
+  flexDirection: 'row',
+};
+
+export const fileHistoryHeaderTitleCss: CSSObject = {
+  fontSize: '20px',
+  fontWeight: 600,
+  margin: 0,
+};
+
+export const fileHeaderInfoWrapperCss: CSSObject = {
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '6px',
+  marginTop: 12,
+};
+
+export const fileHeaderFileInfoCss: CSSObject = {
+  display: 'flex',
+  alignItems: 'center',
+  gap: '6px',
+  color: 'var(--gray-70)',
+  height: '21px',
+};
+
+export const fileHistoryCloseButtonCss: CSSObject = {
+  background: 'none',
+  border: 'none',
+  cursor: 'pointer',
+  display: 'flex',
+  alignSelf: 'flex-start',
+  padding: '0px',
+  '&:hover': {
+    opacity: 0.7,
+  },
+};
+
+export const fileHistoryRestoreCloseButtonCss: CSSObject = {
+  ...fileHistoryCloseButtonCss,
+  alignSelf: 'flex-end',
+};
+
+// Content styles
+export const fileHistoryContentCss: CSSObject = {
+  padding: '6px 0px 6px 16px',
+  overflowY: 'auto',
+};
+
+export const fileHistoryListCss: CSSObject = {
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '16px',
+  position: 'relative',
+};
+
+export const fileHistoryDateHeadingCss: CSSObject = {
+  marginBottom: '8px',
+};
+
+export const fileHistoryTimelineContainerCss: CSSObject = {
+  position: 'relative',
+  ':before': {
+    content: '""',
+    position: 'absolute',
+    left: '20px',
+    top: '15px',
+    bottom: '40px',
+    borderLeft: '1px dashed var(--gray-50)',
+  },
+};
+
+// Version item styles
+export const fileVersionItemWrapperCss: CSSObject = {
+  display: 'flex',
+  gap: '12px',
+  padding: '8px 16px',
+  borderRadius: '8px',
+  ':hover': {
+    backgroundColor: 'var(--gray-20)',
```

Comment:
Add `&:focus-within` along with hover

Then when the user presses Tab, the moment the keyboard focus enters the row:

- The row becomes :focus-within
- Actions appear
- User can tab into the buttons normally

**How it works**

- The button is still focusable even when it’s visually hidden.
- When it receives focus via Tab, the row becomes :focus-within.
- Your :focus-within CSS then shows the actions, so they appear exactly when focus reaches them.